tls/s2n_auth_selection.c

```diff
@@ -177,8 +178,10 @@ int s2n_is_sig_scheme_valid_for_auth(struct s2n_connection *conn, const struct s
     notnull_check(cipher_suite);
 
     GUARD(s2n_certs_exist_for_sig_scheme(conn, sig_scheme));
-    GUARD(s2n_is_sig_alg_valid_for_cipher_suite(sig_scheme->sig_alg, cipher_suite));
-
+    /* We only care about server side signature alg */
```

Comment:
This comment is a little confusing to me. On the server side, we need to ensure that the signature algorithm is compatible with the cert (for now just based on key type since we don't check a match against the signature on the cert) and that the key type is compatible with the cipher suite auth type. On the client side we still care about the signature algorithm. We just need to make sure that the signature algorithm is compatible with the cert, which we do on the line above.

In the short term, I would advocated for tweaking this comment slightly since we do still care about the signature algorithm on the client side. We just don't need it to match the cipher suite auth type.

In the more medium term, we could probably split up this function and/or rename for readability. In the server code path, if we check signature algorithm = cert key type and then separately that cert key type = cipher suite auth type then I'm not sure if it is needed to also check signature algorithm = cipher suite auth type. But maybe it doesn't matter? Just looking for any way to make this code cleaner since it is extremely complex.

Reply:
thanks- cleaned up the wording a bit.